src/Avalonia.Base/Platform/StandardRuntimePlatform.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Avalonia.Platform.Internal;
+
+namespace Avalonia.Platform
+{
+    public class StandardRuntimePlatform : IRuntimePlatform
+    {
+        public IDisposable StartSystemTimer(TimeSpan interval, Action tick)
+        {
+            return new Timer(_ => tick(), null, interval, interval);
+        }
+
+        public IUnmanagedBlob AllocBlob(int size) => new UnmanagedBlob(size);
+        
+        private static readonly Lazy<RuntimePlatformInfo> Info = new(() =>
+        {
+            OperatingSystemType os;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                os = OperatingSystemType.OSX;
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.Create("FREEBSD")))
+                os = OperatingSystemType.Linux;
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                os = OperatingSystemType.WinNT;
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("Android")))
+                os = OperatingSystemType.Android;
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("iOS")))
+                os = OperatingSystemType.iOS;
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("Browser")))
+                os = OperatingSystemType.Browser;
+            else
+                throw new Exception("Unknown OS platform " + RuntimeInformation.OSDescription);
+
+            // Source: https://github.com/dotnet/runtime/blob/main/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs
+            var isCoreClr = Environment.Version.Major >= 5 || RuntimeInformation.FrameworkDescription.StartsWith(".NET Core", StringComparison.OrdinalIgnoreCase);
+            var isMonoRuntime = Type.GetType("Mono.RuntimeStructs") != null;
```

Comment:
I think the traditional way is to check for `Mono.Runtime` - https://www.mono-project.com/docs/gui/winforms/porting-winforms-applications/#runtime-conditionals